ethcore/trace/Cargo.toml

```diff
@@ -0,0 +1,25 @@
+[package]
+description = "Transaction tracing"
+name = "trace"
+version = "0.1.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+
+[dependencies]
+common-types = { path = "../types"}
+ethcore-blockchain = { path = "../blockchain" }
```

Comment:
Do we really need the entire blockchain or just some trait definitions?

Reply:
Currently we use `BlockChainDB` and the contentious `DatabaseExtras`, e.g.:

```
pub struct TraceDB<T> where T: DatabaseExtras {
	/// cache
	traces: RwLock<HashMap<H256, FlatBlockTraces>>,
	/// hashes of cached traces
	cache_manager: RwLock<CacheManager<H256>>,
	/// db
	db: Arc<dyn BlockChainDB>,
	/// tracing enabled
	enabled: bool,
	/// extras
	extras: Arc<T>,
}
```

Reply:
What is your thinking here? Are you thinking we should move trait to own crates?